packages/clay-modal/src/Modal.tsx

```diff
@@ -42,6 +42,11 @@ interface IProps
 	 */
 	observer: Observer;
 
+	/**
+	 * Element to render modal into.
+	 */
+	opener?: Element;
```

Comment:
I think the only thing we could change here is the name of the API, the property can be confused with `window.opener` where the purpose is different, and in this case we just want the container where you want to render the Modal Portal.

Because in this case we would be restricting the API to be used only with `opener={window.opener}` where it makes the most sense but we are creating a very specific API with few use cases, to generalize a little here we can consider `containerElement` or `containerElementRef` the difference is that we can decide that the developer passes the container ref instead of creating it internally.

We may also want to improve our typing here in this case, `Element` will work anyway, but the inference can be a problem sometimes, for example:

```ts
type El = Element;

const shouldReturn = (a: El) => a;

// Returns the type HTMLDivElement
const element = document.createElement('div');

// The inference is that it will return the element with type `Element` and not `HTMLDivElement`.
shouldReturn(element)
```

This will work because `HTMLDivElement` extends from `Element` but we could use generic types to improve type inference.

```ts
type El<T extends Element> = T;

// We still restrict the generic type only to elements that extend from the Element interface.
const shouldReturn = <T extends Element>(a: El<T>) => a;

const element = document.createElement('div');

// The inference here will be that it returns an element of type `HTMLDivElement`
shouldReturn(element)
```

Maybe we can apply this here in our interface but we still have to know if using `React.FunctionComponent<IProps>` can get in the way. Otherwise, we may disregard this for now because it is not a big problem but we may later want to stop using the `FunctionComponent` that has been discouraged of using it lately.